x/evm/precompile/funtoken.go

```diff
@@ -160,13 +160,17 @@ func (p precompileFunToken) bankSend(
 	}
 
 	// Transfer the bank coin
-	err = SafeSendCoinFromModuleToAccount(
+	//
+	// NOTE: The NibiruBankKeeper needs to reference the current [vm.StateDB] before
+	// any operation that has the potential to use Bank send methods. This will
+	// guarantee that [evmkeeper.Keeper.SetAccBalance] journal changes are
+	// recorded if wei (NIBI) is transferred.
+	p.evmKeeper.Bank.StateDB = start.StateDB
```

Comment:
Likewise, `p.evmKeeper.Bank.StateDB` should already be set, so there shouldn't be a need to set it again.